app/models/constraint.rb

```diff
@@ -4,17 +4,24 @@ class Constraint < ApplicationRecord
   has_many :constraint_fields
   has_many :fields, through: :constraint_fields
 
-  enum value_type: %w[Integer String]
+  enum value_type: {
+    integer: 'integer',
+    string:  'string',
+    float:   'float'
+  }, _suffix: true
 
   validates :name, presence: true
   validates_length_of :name, minimum: 4
-  validate :default_is_of_type_value_type
+  validate :default_is_a_value_type
 
   private
-    def default_is_of_type_value_type
+    def default_is_a_value_type
       return if default.nil? || default == 'nil'
-
-      send("default_is_#{value_type.downcase}")
+      begin
+        send("default_is_#{value_type}")
+      rescue
```

Comment:
could you tell me which of the `default_is_*` method can throw an exception? In general an catch-all error-handling should be avoided.

Reply:
The enum has built a in validation but I wanted to avoid an enum error on the front-end.

Reply:
I see. I think instead of catching all types of errors, you can specify to only catch ArgumentError so that other non-validation exceptions are not ignored.

Reply:
Looking back at it, I may have originally come across this due to a bad factory. Trying it in console now, the enum validation happens before the custom validation. Assuming that holds, the rescue will never get called anyway. 

I'll remove the error handline and use [this comment](https://github.com/rails/rails/issues/13971#issuecomment-383905807) as a model.